src/Jackalope/Item.php

```diff
@@ -636,10 +636,6 @@ protected function checkState()
             $this->postDirtyState = -1;
         }
 
-        if ($this->state === self::STATE_DELETED) {
-            throw new InvalidItemStateException('Item '.$this->path.' is deleted');
-        }
```

Comment:
So I guess this implies that `DELETED` is a valid state until you try and manipulate the node.